src/main/java/com/apifest/doclet/Doclet.java

```diff
@@ -82,188 +104,333 @@
+        if (docletMode.contains(DocletMode.DOC) && (mappingDocsOutputFile == null || mappingDocsOutputFile.isEmpty())) {
+            throw new IllegalArgumentException("the mappings docs output file must be provided.");
+        }
+        applicationPath = System.getProperty("application.path");
     }
 
-    private static MappingEndpoint getMappingEndpoint(MethodDoc methodDoc) throws ParseException {
-        MappingEndpoint endpoint = null;
+    private static ParsedEndpoint parseEndpoint(MethodDoc methodDoc) throws ParseException {
+        ParsedEndpoint parsed = null;
+        MappingEndpoint mappingEndpoint = null;
+        MappingEndpointDocumentation mappingEndpointDocumentation = null;
 
         String externalEndpoint = getFirstTag(methodDoc, APIFEST_EXTERNAL);
-        if(externalEndpoint != null){
-            endpoint = new MappingEndpoint();
-            endpoint.setExternalEndpoint("/" + mappingVersion + externalEndpoint);
-
-            String internalEndpoint = getFirstTag(methodDoc, APIFEST_INTERNAL);
-            if (internalEndpoint != null) {
-                if (applicationPath == null || applicationPath.isEmpty() || NULL.equals(applicationPath)) {
-                    endpoint.setInternalEndpoint(internalEndpoint);
-                } else {
-                    endpoint.setInternalEndpoint(applicationPath + internalEndpoint);
-                }
-                Matcher m = VAR_PATTERN.matcher(internalEndpoint);
-                int i = 1;
-                while(m.find()) {
-                    String varName = m.group(2);
-
-                    //get RE if any var in internal path
-                    String varExpression = getFirstTag(methodDoc, APIFEST_RE + varName);
-                    if(varExpression != null) {
-                        if (endpoint.getVarName() == null) {
-                            endpoint.setVarName(varName);
-                            endpoint.setVarExpression(varExpression);
-                        } else {
-                            // add current varName and varExpression with SPACE before that
-                            endpoint.setVarName(endpoint.getVarName() + " " + varName);
-                            endpoint.setVarExpression(endpoint.getVarExpression() + " " + varExpression);
-                        }
-                    }
-                }
+        if (externalEndpoint != null) {
+
+            parsed = new ParsedEndpoint();
+            mappingEndpoint = new MappingEndpoint();
+            mappingEndpointDocumentation = new MappingEndpointDocumentation();
+
+            mappingEndpoint.setExternalEndpoint("/" + mappingVersion + externalEndpoint);
+            mappingEndpointDocumentation.setEndpoint("/" + mappingVersion + externalEndpoint);
+
+            parseInternalEndpointTag(methodDoc, mappingEndpoint);
+            parseDocsGroupTag(methodDoc, mappingEndpointDocumentation);
+            parseDocsSummaryTap(methodDoc, mappingEndpointDocumentation);
```

Comment:
Please, rename to parseDocsSummaryTag (G at the end instead of P ;)) and have a beer on tap :)